### ISSUE TITLE
Reporting Enhancements

### ISSUE THREAD

**Issue by klgrimley** [bmlt-enabled/yap#574]
Having recently implemented YAP our helpline chair asked about more/improved reporting.
Some of this would be for presenting data to the service body. And some would help to improve volunteer scheduling.

- The ability to set a date range for viewing/printing report logs
- The ability to see at a glance the number of calls to volunteers received for a given date range
- The ability to see at a glance the number of meeting sms requests for a given date range
- The ability to see at a glance  the number and time of day of missed calls for a given date range

**Comment by dgershman** [comment]
Going to add a date-picker as the first step here.

**Comment by EzraKuhr** [comment]
+1 to this request.

**Comment by mczelusniak** [comment]
I've had a lot of areas (FL Region) requesting this as well.  Especially "the ability to see at a glance the number of meeting sms requests for a given date range"
Thank you!

**Comment by dgershman** [comment]
<img width="1173" alt="image" src="https://user-images.githubusercontent.com/1850811/172025015-39e29456-f477-41cf-a02e-4fb445099f3d.png">

**Comment by dgershman** [comment]
Rough version of summary 

<img width="1165" alt="image" src="https://user-images.githubusercontent.com/1850811/172029691-58c91439-05ee-4afb-af6d-18e09ab1cccd.png">

**Comment by alanb2718** [comment]
This will be useful for us as well!  One small but helpful addition to this summary would be to report the number of completed calls to a volunteer, and the number of attempted calls to a volunteer that went unanswered.

**Comment by dgershman** [comment]
I'm sure there are number of other metrics that could be added over time.  Everyone will want their pet metric!

**Comment by kgrimley-bu** [comment]
Agree with @alanb2718 those are metrics our Area are interested in. Also, part of the original ticket. No rush, just something we'd like to see. Thanks for adding the custom range. That's a huge help for us!

**Comment by pjaudiomv** [comment]
mine too, I currently use some crazy php script i wrote to parse and get those metrics

**Comment by dgershman** [comment]
First pass at reporting metric summaries.  I've included two basic ones here:

<img width="1018" alt="image" src="https://user-images.githubusercontent.com/1850811/180653211-065f8a50-2f6b-4366-9e00-c85f21cb137a.png">

**Comment by dgershman** [comment]
Currently (as of the latest stable release 4.0.4), no events are captured as to whether a meeting was searched by phone or SMS, so historical data will be difficult if not impossible to find.  This will be a going forward metric point.

**Comment by dgershman** [comment]
SMS Metrics + Reporting included now
![Screen Shot 2022-07-26 at 4 04 47 PM](https://user-images.githubusercontent.com/1850811/181102261-958f82c9-ecb6-441f-90ef-daeae1aaa02d.png)
![Screen Shot 2022-07-26 at 4 06 13 PM](https://user-images.githubusercontent.com/1850811/181102373-b93fb015-4e95-4939-92ff-0d1c4a625159.png)

**Comment by dgershman** [comment]
Almost all of these are Available in 4.1.0-beta2 if you want to test. 

https://github.com/bmlt-enabled/yap/releases/tag/4.1.0-beta2

**Comment by klgrimley** [comment]
I'll install and let you know if there are any issues. Thanks for the work on this

**Comment by klgrimley** [comment]
After initial look seems to working as expected. I'll get with our Helpline chair to go over the specifics. Am getting one "error" but can't tell if it's having any practical effect on reporting.
```Remote Pagination Error - Server response missing 'last_page' property```
```

O._parseRemoteData | @ | yap-reports.min.js:formatted:12841
-- | -- | --
  | (anonymous) | @ | yap-reports.min.js:formatted:12829
  | Promise.then (async) |   |  
  | (anonymous) | @ | yap-reports.min.js:formatted:12828
  | O._getRemotePage | @ | yap-reports.min.js:formatted:12811
  | (anonymous) | @ | yap-reports.min.js:formatted:12795
  | O.trigger | @ | yap-reports.min.js:formatted:12782
  | (anonymous) | @ | yap-reports.min.js:formatted:12656
  | O.setPage | @ | yap-reports.min.js:formatted:12654
  | u._setData | @ | yap-reports.min.js:formatted:5971
  | u.setData | @ | yap-reports.min.js:formatted:5966
  | updateCallRecords | @ | yap.min.js:1
  | updateAllReports | @ | yap.min.js:1
  | (anonymous) | @ | yap.min.js:1
  | dispatch | @ | yap.min.js:1
  | g.handle | @ | yap.min.js:1

```

**Comment by pjaudiomv** [comment]
I also got this same error and wasn't sure what to make of it

**Comment by dgershman** [comment]
Was there any repro steps?

From: Kevin Grimley ***@***.***>
Date: Monday, August 1, 2022 at 4:40 PM
To: bmlt-enabled/yap ***@***.***>
Cc: Danny Gershman ***@***.***>, Comment ***@***.***>
Subject: Re: [bmlt-enabled/yap] Reporting Enhancements (Issue #574)

After initial look seems to working as expected. I'll get with our Helpline chair to go over the specifics. Am getting one "error" but can't tell if it's having any practical effect on reporting.
Remote Pagination Error - Server response missing 'last_page' property





O._parseRemoteData | @ | yap-reports.min.js:formatted:12841



-- | -- | --



  | (anonymous) | @ | yap-reports.min.js:formatted:12829



  | Promise.then (async) |   |



  | (anonymous) | @ | yap-reports.min.js:formatted:12828



  | O._getRemotePage | @ | yap-reports.min.js:formatted:12811



  | (anonymous) | @ | yap-reports.min.js:formatted:12795



  | O.trigger | @ | yap-reports.min.js:formatted:12782



  | (anonymous) | @ | yap-reports.min.js:formatted:12656



  | O.setPage | @ | yap-reports.min.js:formatted:12654



  | u._setData | @ | yap-reports.min.js:formatted:5971



  | u.setData | @ | yap-reports.min.js:formatted:5966



  | updateCallRecords | @ | yap.min.js:1



  | updateAllReports | @ | yap.min.js:1



  | (anonymous) | @ | yap.min.js:1



  | dispatch | @ | yap.min.js:1



  | g.handle | @ | yap.min.js:1

**Comment by pjaudiomv** [comment]
for me with firefox this happens just by clicking on reports tab then selecting a service body no additional steps

**Comment by klgrimley** [comment]
Same for me in Chrome

**Comment by dgershman** [comment]
resolved in beta3.

**Comment by dgershman** [comment]
fixes released in 4.1.0

**Comment by dgershman** [comment]
Split off the missed call count feature to it's own ticket https://github.com/bmlt-enabled/yap/issues/611

**Comment by dgershman** [comment]
> This will be useful for us as well!  One small but helpful addition to this summary would be to report the number of completed calls to a volunteer, and the number of attempted calls to a volunteer that went unanswered.

This is essentially addressed in 4.1.1

**Comment by dgershman** [comment]
> Split off the missed call count feature to it's own ticket https://github.com/bmlt-enabled/yap/issues/611

Addressed in 4.1.1 which was released a few moments ago.
